GitHubEvents.ps1

```diff
@@ -0,0 +1,125 @@
+# Licensed under the MIT License.
+
+function Get-GitHubEvent
+{
+<#
+    .DESCRIPTION
+        Lists events for an issue, repository, or a single event
+
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .PARAMETER OwnerName
+        Owner of the repository.
+        If not supplied here, the DefaultOwnerName configuration property value will be used.
+
+    .PARAMETER RepositoryName
+        Name of the repository.
+        If not supplied here, the DefaultRepositoryName configuration property value will be used.
+
+    .PARAMETER Uri
+        Uri for the repository.
+        The OwnerName and RepositoryName will be extracted from here instead of needing to provide
+        them individually.
+
+    .PARAMETER EventID
+        The ID of a specific event to get. If not supplied, will return back all events for this repository.
+
+    .PARAMETER Issue
+        Issue number to get events for. If not supplied, will return back all events for this repository.
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .EXAMPLE
+        Get-GitHubEvent -OwnerName Powershell -RepositoryName PowerShellForGitHub
+
+        Get the events for the PowerShell\PowerShellForGitHub project.
+#>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        DefaultParametersetName='RepositoryElements')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSShouldProcess", "", Justification="Methods called within here make use of PSShouldProcess, and the switch is passed on to them inherently.")]
+    param(
+        [Parameter(Mandatory, ParameterSetName='RepositoryElements')]
+        [Parameter(Mandatory, ParameterSetName='IssueElements')]
+        [Parameter(Mandatory, ParameterSetName='EventElements')]
+        [string] $OwnerName,
+
+        [Parameter(Mandatory, ParameterSetName='RepositoryElements')]
+        [Parameter(Mandatory, ParameterSetName='IssueElements')]
+        [Parameter(Mandatory, ParameterSetName='EventElements')]
+        [string] $RepositoryName,
+
+        [Parameter(Mandatory, ParameterSetName='RepositoryUri')]
+        [Parameter(Mandatory, ParameterSetName='IssueUri')]
+        [Parameter(Mandatory, ParameterSetName='EventUri')]
+        [string] $Uri,
+
+        [Parameter(Mandatory, ParameterSetName='EventUri')]
+        [Parameter(Mandatory, ParameterSetName='EventElements')]
+        [string] $EventID,
```

Comment:
I think this should be `Event` to be consistent with how we name the other parameters.

Reply:
We chose not to use `$Event` because it is an existing PowerShell automatic variable.